syntax = "proto3";

option java_package = "com.inforeach.eltrader.tms.api.grpc";

// Common data types

message Void {
}

message Fields {
    map<string, string> stringFields = 1;
    map<string, double> numericFields = 2;
}

message FIXFields {
    map<int32, string> stringFields = 1;
    map<int32, double> numericFields = 2;
}

enum OrdType {
    OrdType_Undefined=0;
    OrdType_Market=49;
    OrdType_Limit=50;
    OrdType_Stop=51;
    OrdType_Stop_Limit=52;
}

enum Side {
    Side_Undefined=0;
    Side_Buy=49;
    Side_Sell=50;
    Side_Buy_Minus=51;
    Side_Sell_Plus=52;
    Side_Sell_Short=53;
}

enum WaveSizeType {
    WaveSizeType_PctTgtQty=0;
    WaveSizeType_Qty=1;
    WaveSizeType_Value=2;
}

enum HandlInst {
    HandlInst_Undefined=0;
    HandlInst_AutomatedExecutionOrderPrivateNoBrokerIntervention=49;
    HandlInst_AutomatedExecutionOrderPublicBrokerInterventionOk=50;
    HandlInst_ManualOrderBestExecution=51;
}


enum FIXTag {
    FIXTag_Undefined=0;
    FIXTag_Account=1;
    FIXTag_HandlInst=21;
    FIXTag_Symbol=55;
    FIXTag_Side=54;
    FIXTag_OrdType=40;
    FIXTag_OrderQty=38;
    FIXTag_Price=44;
    FIXTag_Text=58;
    FIXTag_LastPx=31;
    FIXTag_LastQty=32;
    FIXTag_TrnDestination=16501;
    FIXTag_Instrument=16505;
    FIXTag_SingleOrderTransactionId=17700;
}